source/common/http/http2/metadata_decoder.h

```diff
@@ -58,18 +59,15 @@ class MetadataDecoder : Logger::Loggable<Logger::Id::http2> {
    */
   bool decodeMetadataPayloadUsingNghttp2(bool end_metadata);
 
+  // Metadata that is currently under decoding.
+  std::unique_ptr<MetadataMap> metadata_map_;
```

Comment:
nit: MetadataMapPtr

Reply:
+1, plus "is currently under decoding." -> "is currently being decoded."?